## Ką mokomės

||||
|:----:|:-----:|:------:|
|![](sam1.png)|![](sam2.png)|![](sam3.png)|

||||
|:----:|:-----:|:------:|
|![](int1.png)|![](int2.png)|![](int3.png)|

## Užrašai
Užrašuose yra 5 sąsiuvinio nuotraukos su Irmanto paskaitų sprendimais. Jos padės suprasti, kokias integravimo metodikas naudoja per paskaitas.

||||||
|:----:|:-----:|:------:|:-----:|:------:|
|![](notes1.png)|![](notes2.png)|![](notes3.png)|![](notes4.png)|![](notes5.png)|

## Šį susitikimą sprendžiame
Irmantui: turėk atsidaręs tas pačias nuotraukas pas save, nes man netelpa atsakymai

||||
|:----:|:-----:|:------:|
|![](ribos1.jpg)|![](ribos2.jpg)|![](ribos3.jpg)|

## Ribų tipai

## Tigrų lenktynės
Lenktyniauja du tigrai. Galimos trys baigtys:
    
|Pirmas galingesnis >>>>>>>>>>>>>>>>| Antras galingesnis >>>>>>>>>>>>>>>>| Fotofinišas - reik žiūrėt kofus >>>>>>>>>>>>>>>>|
|:---:|:---:|:---:|
|$\displaystyle\lim_{x\to+\infty}\frac{10n^4-7n+1}{4n^3-3n^2+10n-4}=\frac{+\infty}{1}$|$\displaystyle\lim_{n \rightarrow \infty} \frac{\sqrt[3]{n^{2}+2n-1}+\sqrt{n}}{\sqrt[4]{n^{3}+n^{2}+1}-\sqrt{n-1}}=\frac{1}{+\infty}$|$\displaystyle\lim_{x\to+\infty}\frac{8n^3+3n^2+5}{6-7n^2+10n^3} = \frac{8}{10}$|

Po to atsiranda papildomų momentų:
* Jei nori patikrint, kodėl vienas ar kitas laimės, tai dalini abu iš didžiausio surasto laipsnio:
  
  $\displaystyle\lim_{x\to+\infty}\frac{8n^3+3n^2+5}{6-7n^2+10n^3}=\lim_{x\to+\infty}\frac{8n^3/n^3+3n^2/n^3+5/n^3}{6/n^3-7n^2/n^3+10n^3/n^3} = \lim_{x\to+\infty}\frac{8+3/n+5/n^3}{6/n^3-7/n+10}=\frac{8}{10}$
* Remiantis panašiomis idėjomis, kai skaičiuoji ribą $\displaystyle\lim_{x\to+\infty}\frac{8^n+3^n}{9^n-5^n}$ irgi gali panašiai surasti galingiausią narį ir iš jo padalinti likusius narius.
* **Išvada 1.** Jei nori įvertint reiškinių ribą, pvz. šaknies iš kito reiškinio ribą, tai galima panašiai naikinti nykstamai mažus narius.

  Kai $x\to+\infty$, tai $\sqrt{4x^3+x+1} \to \sqrt{4x^3} = 2x^{3/2}$
* **Išvada 2.** Nykstamus narius galime naikinti visada vos tik pamatome, kad vyksta tigrų lenktynės.

**Pavydys 1.** $\displaystyle\lim_{n \rightarrow \infty} \frac{\sqrt{n+2}+\sqrt[3]{n+1}}{\sqrt[4]{n^{2}-n+3}-\sqrt[5]{n^2}}$

**Pavydys 2.**$\displaystyle\lim_{n \rightarrow \infty} \frac{2^n-4^n}{4^{n+2}-3^n}$

$\displaystyle \frac{\sqrt{n+2}+\sqrt[3]{n+1}}{\sqrt[4]{n^{2}-n+3}-\sqrt[5]{n^2}} \to \frac{\sqrt{n}+\sqrt[3]{n}}{\sqrt[4]{n^{2}}-\sqrt[5]{n^2}} \to \frac{\sqrt{n}}{\sqrt[4]{n^{2}}}$, kai $n \to +\infty$

$\displaystyle \frac{\sqrt{n/n+2/n}+\sqrt[3]{n/n^{3/2}+1/n^{3/2}}}{\sqrt[4]{n^{2}-n+3}-\sqrt[5]{n^2}}$

$\sqrt[3]{n} / \sqrt{n} = n^{{1/3} - {1/2}}=n^{-1/6}\to 0$

### Konjunkcijos
* Konjunkcijos? Tai [ne dangaus kūnų](https://en.wikipedia.org/wiki/Conjunction_(astronomy)), o greičiau lygybės $ab=c$ kitų formų tarpusavio ryšys. 
* Tiksliau lygybių $\boxed{(a-b)(a+b)=a^2-b^2}$, $\boxed{(a-b)(a^2+ab+b^2)=a^3-b^3}$ ir $\boxed{(a+b)(a^2-ab+b^2)=a^3+b^3}$
* $(a+b)(a-b) = \begin{array}{c||c|c} & a & b \\ \hline a & a^2 & ab \\ \hline -b & -ab & -b^2 \end{array} = a^2-b^2$
* $(a+b)(a^2-ab+b^2) = \begin{array}{c||c|c} & a & b \\ \hline a^2 & a^3 & a^2b \\ \hline -ab & -a^2b & -ab^2 \\ \hline b^2 & ab^2 & b^3 \end{array} = a^3+b^3$
* $(a-b)(a^2+ab+b^2) = \begin{array}{c||c|c} & a & -b \\ \hline a^2 & a^3 & -a^2b \\ \hline ab & a^2b & -ab^2 \\ \hline b^2 & ab^2 & -b^3 \end{array} = a^3-b^3$

**Pavyzdys 1.** Kai skaičiuoji $\displaystyle\lim_{n\to+\infty} \left(\sqrt[3]{n^3-1} - n\right)\cdot n$, tai paimk $\displaystyle a-b=\frac{a^3-b^3}{a^2+ab+b^2}$ ir įstatyk $\begin{cases} a \to \sqrt[3]{n^3-1} \\ b \to n\end{cases}$

Gausi $\displaystyle \sqrt[3]{n^3-1}-n=\frac{n^3-1-n^3}{\sqrt[3]{(n^3-1)^2}+n\sqrt[3]{n^3-1}+n^2}$. Po to vyks tigrų lenktynės.

**Pavyzdys 2.** Kai skaičiuoji $\displaystyle\lim_{n\to+2} \frac{\sqrt{n} - \sqrt{2}}{n-2}$, tai paimk $\displaystyle a-b=\frac{a^2-b^2}{a+b}$ ir įstatyk $\begin{cases} a \to \sqrt{n} \\ b \to \sqrt{2}\end{cases}$

Gausi $\displaystyle \frac{\sqrt{n} - \sqrt{2}}{n-2} = \frac{(n - 2)/(\sqrt{n} + \sqrt{2})}{n-2} = \frac{1}{\sqrt{n} + \sqrt{2}}$. 

**Pavyzdys 3.** Kai skaičiuoji $\displaystyle\lim_{n\to+\infty} \frac{(n-1)^3+(n+3)^3}{(n-2)^4-(n+4)^4}$, tai paimk $\displaystyle a^4-b^4=(a-b)(a^3+a^2b+ab^2+b^3)$ ir įstatyk $\begin{cases} a \to n-2 \\ b \to n+4\end{cases}$. Vėliau tigrų lenktynės, naikiname nykstamai mažus narius ir gauname atsakymą.

Gausi $\displaystyle (n-2)^4-(n+4)^4=-6\left((n-2)^3+(n-2)^2(n+4)+(n-2)(n+4)^2+(n+4)^3\right)$.

**Pavyzdys 4.** $\displaystyle\lim_{n\to+3} \frac{\sqrt{x}+\sqrt{x-3}-\sqrt{3}}{\sqrt{x^2-9}}$

**Pavyzdys 5.** $\displaystyle\lim_{n\to+3} \frac{\sqrt[3]{x-30}+3}{x^3-27}$

### Dalyba kampu
* Jei $P(a)=0$, tai $P(x)$ dalijasi iš $x-a$
* Kitais žodžiais, jei 
$$a_0a^n+a_{1}a^{n-1}+a_{2}a^{n-2}+\dots a_{n-2}a^2+a_{n-1}a+a_n = 0\text{, tai }$$
$$a_0x^n+a_{1}x^{n-1}+a_{2}x^{n-2}+\dots a_{n-2}x^2+a_{n-1}x+a_n\text{, dalijasi iš }x-a$$
* Įrodymas: 
  * $P(x) = P(x)-P(a) = a_0(x^n-a^n)+a_{1}(x^{n-1}-a^{n-1})+a_{2}(x^{n-2}-a^{n-2})+\dots a_{n-2}(x^2-a^2)+a_{n-1}(x-a) = 0$
  * Kiekvienas dėmuo dalijasi iš $x-a$
  
**Dalybos pavyzdys** 

<img src="kampas.png" width="400">

### Mistinis vieneto prieaugis

Niutono laikais (prieš 400m.), kai buvo įvestas ribos žymėjimas, buvo stengiamasi kiekvienam reiškiniui surasti metodiką, leidžiančią apskaičiuoti to reiškinio ribą. Tačiau numatyti visas galimas technikas reikėtų labai daug žinių. Kaip pavyzdį mokinomės pagrįsti, kodėl $\sin x\to x$, kai $x \to 0$ ir sprendimas buvo specifinis, nematytas jokiuose kituose uždaviniuose (reikėjo suvokti, kad vienetiniame skritulyje pusė lanko ($x$) artėja į pusę stygos ($\sin x$), kai mažiname išpjovos kampą $x$). Čia pasitarnauja Lopitalio atrasta taisyklė, su kuria galime apskaičiuoti labai daug ribų: 

* Jei $x\to a$ ir $f(x)$ kartu su $g(x)$ artėja į 0 arba $f(x)$ kartu su $g(x)$ artėja į $+\infty$, tai $\displaystyle\lim_{x\to a} \frac{f(x)}{g(x)} = \lim_{x\to a} \frac{f'(x)}{g'(x)}$

Nors ši taisyklė ir galingas įrankis, tačiau ne visuomet veikia lengvai. Kartais reikia numatyti kelias formas, kaip pertvarkyti reiškinį.Pvz:

* $\displaystyle\lim_{x\to +\infty} xe^{-x}=\lim_{x\to +\infty} \frac{e^{-x}}{1/x} = \lim_{x\to +\infty} \frac{-e^{-x}}{-1/x^2}=\lim_{x\to +\infty} \frac{e^{-x}}{2/x^3}=\lim_{x\to +\infty} \frac{-e^{-x}}{-6/x^4}=\dots$, todėl patenkame į niekad nesibaigiantį procesą. Bet:
* $\displaystyle\lim_{x\to +\infty} xe^{-x}=\lim_{x\to +\infty} \frac{x}{e^x} = \lim_{x\to +\infty} \frac{1}{e^x}=0$ - pavyksta.

Tarp ribų, kurios nepasiduoda jokioms ankstesnėms technikoms ir netgi Lopitaliui, atsiranda viena populiari situacija.

In [51]:
from IPython.display import display, Latex, Markdown
import numpy as np

for x in [1,2,3,5,10,20,50,100,200,500,1000,1000000]:
    display(Markdown(fr'Kai $x={x}$, tai $\left(1+\frac{{{1}}}{{{"x"}}}\right)^x = \left({1}+\frac{{{1}}}{{{x}}}\right)^{{{x}}} = {round((1+1/x)**x, 10)}$'))
    
display(Markdown(fr'Kai $x \to +\infty$, tai $\left(1+\frac{{{1}}}{{{"x"}}}\right)^x \to {np.e}$'))

Kai $x=1$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{1}\right)^{1} = 2.0$

Kai $x=2$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{2}\right)^{2} = 2.25$

Kai $x=3$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{3}\right)^{3} = 2.3703703704$

Kai $x=5$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{5}\right)^{5} = 2.48832$

Kai $x=10$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{10}\right)^{10} = 2.5937424601$

Kai $x=20$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{20}\right)^{20} = 2.6532977051$

Kai $x=50$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{50}\right)^{50} = 2.6915880291$

Kai $x=100$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{100}\right)^{100} = 2.7048138294$

Kai $x=200$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{200}\right)^{200} = 2.7115171229$

Kai $x=500$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{500}\right)^{500} = 2.7155685207$

Kai $x=1000$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{1000}\right)^{1000} = 2.7169239322$

Kai $x=1000000$, tai $\left(1+\frac{1}{x}\right)^x = \left(1+\frac{1}{1000000}\right)^{1000000} = 2.7182804691$

Kai $x \to +\infty$, tai $\left(1+\frac{1}{x}\right)^x \to 2.718281828459045$

In [52]:
for x in [2,3,5,10,20,50,100,200,500,1000,1000000]:
    display(Markdown(fr'Kai $x={x}$, tai $\left(1+\frac{{{1}}}{{{"-x"}}}\right)^{{{"-x"}}} = \left({1}+\frac{{{1}}}{{{-x}}}\right)^{{{-x}}} = {round((1-1/x)**(-x), 10)}$'))
    
display(Markdown(fr'Kai $x \to +\infty$, tai $\left(1+\frac{{{1}}}{{{"-x"}}}\right)^{{{"-x"}}} \to {np.e}$'))

Kai $x=2$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-2}\right)^{-2} = 4.0$

Kai $x=3$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-3}\right)^{-3} = 3.375$

Kai $x=5$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-5}\right)^{-5} = 3.0517578125$

Kai $x=10$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-10}\right)^{-10} = 2.8679719908$

Kai $x=20$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-20}\right)^{-20} = 2.7895098175$

Kai $x=50$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-50}\right)^{-50} = 2.7459727009$

Kai $x=100$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-100}\right)^{-100} = 2.7319990264$

Kai $x=200$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-200}\right)^{-200} = 2.7251088294$

Kai $x=500$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-500}\right)^{-500} = 2.7210051033$

Kai $x=1000$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-1000}\right)^{-1000} = 2.7196422164$

Kai $x=1000000$, tai $\left(1+\frac{1}{-x}\right)^{-x} = \left(1+\frac{1}{-1000000}\right)^{-1000000} = 2.7182831877$

Kai $x \to +\infty$, tai $\left(1+\frac{1}{-x}\right)^{-x} \to 2.718281828459045$

Kaip įprasta matematikoje, naujam, niekaip prieš tai matytais metodais nepaaiškinamam skaičiui rezervuojamas atskiras žymėjimas (panašiai atsitiko su šaknies, laipsnio, skaičių $i$ ir $\pi$ žymėjimais). Abejose situacijose galima tvirtinti, kad:

Kai $x\to \infty$, tai $\left(1+\frac{1}{x}\right)^x \to e$, kur $e\approx 2.718281828$

Šią taisyklę galime suformuluoti ir dar paprasčiau:

Jei tik $a\to 0$ ir $ab=1$, tai $(1+a)^b\to e$

**Pavyzdys** $\displaystyle\lim_{x\to\infty} \left(\frac{x^2+3}{x^2+1}\right)^{4x^2} = \lim_{x\to\infty} \left(\frac{x^2+1}{x^2+1}+\frac{2}{x^2+1}\right)^{4x^2} = \lim_{x\to\infty} \left(1+\frac{2}{x^2+1}\right)^{4x^2} = \lim_{x\to\infty} \left(\left(1+\frac{2}{x^2+1}\right)^\frac{x^2+1}{2}\right)^{\frac{2}{x^2+1}\cdot 4x^2} = \lim_{x\to\infty} e^{\frac{2}{x^2+1}\cdot 4x^2} = e^8$

Skaičiuojant šią ribą reikėjo pastebėti, kad $\displaystyle\frac{2}{x^2+1}\to 0$, kai $x\to +\infty$ ir $\displaystyle\frac{2}{x^2+1} \times \frac{x^2+1}{2}=1$

### Papildomos taisyklės

* $\sin(x) \to x$, kai $x\to 0$
* $\arcsin(\sin x) = x$
* $(a^x)'=\ln(a)\cdot a^x$
* $\displaystyle\frac{a^x-1}{x} \to \ln(a)$, kai $x\to 0$ (Pagal Lopitalį). Kita forma $a^x \to x\ln(a)+1,$ kai $x\to 0$
* $\sin^2(x)+\cos^2(x) = 1$
* $\frac{\sin(x)}{\cos(x)}=\text{tg}(x)$
* [Lopitalio taisyklė](https://en.wikipedia.org/wiki/L%27Hôpital%27s_rule)
* Skaidymas dauginamaisiais grupavimo būdu (jei nepatinka, dalyba kampu iš $t-\sqrt{3}$)

### Keverzonės apie sudėtinių išvestinių skaičiavimą

$\displaystyle e^{-x^2/2}\cdot \ln(x) = \frac{e^{-x^2/2}}{1/\ln(x)} = \frac{\ln(x)}{1/e^{-x^2/2}} = \frac{\ln(x)}{e^{x^2/2}}$

$f(g(x))' = f'(g(x))\cdot g'(x)$

$\begin{cases} f(x) = e^x \\ g(x) = x^2/2 \\ f'(x)=e^x \\ g'(x) = x\end{cases}$

$f'(g(x))=e^{x^2/2}$ ir $g'(x)=x$

$f'(g(x))\cdot g'(x)=x = e^{x^2/2}\cdot x$